AdminWk4/src/main/java/com/lms/LMSAdmin/controller/AuthorController.java

```diff
@@ -0,0 +1,100 @@
+package com.lms.LMSAdmin.controller;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Produces;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
+
+import com.lms.LMSAdmin.pojo.Author;
+import com.lms.LMSAdmin.service.AuthorService;
+
+@RestController
+@RequestMapping("/admin/authors")
+@Produces({"application/xml", "application/json"})
+@Consumes({"application/xml", "application/json"})
+public class AuthorController {
+
+	@Autowired
+	AuthorService authorService;
+	
+	@ExceptionHandler(HttpClientErrorException.class)
+	@ResponseStatus(HttpStatus.NOT_FOUND)
+	public @ResponseBody String handleResourceNotFound() {
+		return "Invalid resource.";
+	}
+	
+	@ExceptionHandler(MethodArgumentTypeMismatchException.class)

```

Comment:
Good job on catching String exceptions inside the URI request but we need to be able catch them inside the JSON body too. I'll be working on that too and hopefully we will have a solution for that soon.